ReactWindows/ReactNative.Net46/Bridge/DispatcherHelpers.cs

```diff
@@ -151,20 +151,82 @@ public static Task<T> CallOnDispatcher<T>(Dispatcher dispatcher, Func<T> func, b
         {
             if (allowInlining && IsOnDispatcher(dispatcher))
             {
-                return Task.FromResult(func());
+                try
+                {
+                    var result = func();
+                    return Task.FromResult(result);
+                }
+                catch (Exception e)
```

Comment:
This catch is not needed. If `func` causes an exception, it will be thrown to the caller.

Reply:
That's true, though the return type of Task<T> could also be interpreted as "I want the exception to be in there in the task as well, not just the result".
This is done automatically inside the compiled code for method if "async" is specified.
Now "async" is not specified here, so this can be simulated as Matt did. Not sure what the expectation is outside (does the "async" keyword carries some informal expectation for the caller at all?)

Anyway, that Task<T> is awaitable, so it'd be nice for the potential await to behave consistently with the other cases (so it gets result or throws further).

But yeah, gray area.


---
In reply to: [195434819](https://github.com/Microsoft/react-native-windows/pull/1867#discussion_r195434819) [](ancestors = 195434819)